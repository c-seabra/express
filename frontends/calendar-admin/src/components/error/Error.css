.error {
  background-color: red;
  color: white;
  margin: 10px 0;
  padding: 5px;
}
