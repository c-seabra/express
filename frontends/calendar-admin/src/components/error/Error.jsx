import styles from './Error.css'
import React from 'react'

const Error = ({ errors }) => (errors && errors.length > 0 ? <div className={styles.error}>{errors.join(', ')}</div> : null)

export default Error
