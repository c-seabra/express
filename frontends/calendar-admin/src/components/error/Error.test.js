import React from 'react';
import renderer from 'react-test-renderer';
import Error from './Error.jsx';

describe('Error', () => {
  test('renders single error message', () => {
    const component = renderer.create(
      <Error errors={['Bad request']} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders two error messages', () => {
    const component = renderer.create(
      <Error errors={['Bad request', 'Fetch failed']} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('with empty props doesn\'t render', () => {
    const component = renderer.create(
      <Error errors={[]} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('without props doesn\'t render', () => {
    const component = renderer.create(
      <Error />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
})
