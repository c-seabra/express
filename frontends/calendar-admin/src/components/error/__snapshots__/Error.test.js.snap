// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders single error message 1`] = `
<div
  className={undefined}
>
  Bad request
</div>
`;

exports[`Error renders two error messages 1`] = `
<div
  className={undefined}
>
  Bad request, Fetch failed
</div>
`;

exports[`Error with empty props doesn't render 1`] = `null`;

exports[`Error without props doesn't render 1`] = `null`;
